enderio-machines/src/main/java/com/enderio/machines/common/blocks/fluid_tank/FluidTankBlockEntity.java

```diff
@@ -338,7 +337,8 @@ private void tryMendTool() {
             var enchantmentsRecipe = level.registryAccess().lookupOrThrow(Registries.ENCHANTMENT);
             var mendingEnchantment = enchantmentsRecipe.getOrThrow(Enchantments.MENDING);
 
-            if (tool.isDamageableItem() && tool.getEnchantmentLevel(mendingEnchantment) > 0) {
+            if (tool.isDamageableItem() && tool.getEnchantmentLevel(mendingEnchantment) > 0
+                    && FLUID_DRAIN_OUTPUT.getItemStack(this).isEmpty()) {
```

Comment:
Looks good! Can you move this check to the top before the registry lookups?

Reply:
Done